dash/quorum/validator_conn_executor.go

```diff
@@ -241,26 +242,23 @@ func (vc *ValidatorConnExecutor) resolveNodeID(va *types.ValidatorAddress) error
 	if va.NodeID != "" {
 		return nil
 	}
-	var allErrors string
-	for _, resolver := range vc.nodeIDResolvers {
+	var allErrors error
+	for method, resolver := range vc.nodeIDResolvers {
```

Comment:
since, an iteration order is important (because map iteration is not deterministic), then you need to prepare the orders list of keys for it
therefore, it would be easier to define constants for keys and use them in a loop.
```
const(
    DashDialer = "DashDialer"
    TCPNodeIDResolver = "TCPNodeIDResolver"
)
```
and iterate them
```
for _, method  := range []string{DashDialer, TCPNodeIDResolver} {
    err = vc.nodeIDResolvers[method](*va)
    ....
}
```